src/sentry/static/sentry/app/components/globalSelectionLink.jsx

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Link as RouterLink} from 'react-router';
+import {pick} from 'lodash';
+
+import {URL_PARAM} from 'app/components/organizations/globalSelectionHeader/constants';
+
+/**
+ * A modified link used for navigating between organization level pages that
+ * will keep the global selection values (projects, environments, time) in the
+ * querystring when navigating if it's present
+ *
+ * Falls back to <a> if there is no router present.
+ */
+export default class GlobalSelectionLink extends React.Component {
+  static propTypes = {
+    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
+  };
+
+  static contextTypes = {
+    location: PropTypes.object,
+  };
+
+  render() {
+    const {location} = this.context;
+
+    const query = pick(location.query, Object.values(URL_PARAM));
```

Comment:
I recently added a utility function for this. https://github.com/getsentry/sentry/blob/d130c1b66c87943affb985ef9ada75a1915ccce4/src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/utils.jsx#L50-L52

It also removes empty/null values.

Reply:
👍